derp/derp_server.go

```diff
@@ -1560,21 +1560,23 @@ func (s *Server) AddPacketForwarder(dst key.NodePublic, fwd PacketForwarder) {
 			// Duplicate registration of same forwarder. Ignore.
 			return
 		}
-		if m, ok := prev.(multiForwarder); ok {
-			if _, ok := m[fwd]; ok {
+		if m, ok := prev.(*multiForwarder); ok {
+			if m.exists(fwd) {
```

Comment:
so I think the prior assumption was all these multiForwarders were guarded by s.mu.

Where is that not the case? I didn't see the full race output.

Reply:
Annoyingly, I could not capture stack traces of all goroutines before they expired from systemd journal, but I suspect what happened here was a map write concurrent with this [ForwardPacket](https://github.com/tailscale/tailscale/blob/0f7da5c7dc0ae30f02320ede18d20473ad7ca981/derp/derp_server.go#L933) that needs to iterate over the map to find the lowest id.